src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -252,17 +252,20 @@ private void GenerateCompilandsAndLog(BinaryAnalyzerContext context, Dictionary<
                 compilandsWithOneOrMoreInsecureFileHashes.Remove(HashType.None);
             }
 
-            string[] messages = new string[compilandsWithOneOrMoreInsecureFileHashes.Count];
-
-            int hashTypeCount = 0;
-            foreach (HashType hashType in compilandsWithOneOrMoreInsecureFileHashes.Keys)
+            if (compilandsWithOneOrMoreInsecureFileHashes.Count > 0)
```

Comment:
>> 0 [](http://example.com/codeflow?start=64&length=3)

when all error are caused by no hash, the current code will also generate an error with no item in the list.
generate an error will cause pipeline to fail, so we should fix it.
No test was added because the fix is clear, let me know I can add tests.

Reply:
@shaopeng-gh The problem is unclear. Could you add an example to the PR description please?

Reply:
will work on test and update the PR description.